SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc

```diff
@@ -43,6 +43,117 @@ HcalTriggerPrimitiveAlgo::HcalTriggerPrimitiveAlgo( bool pf, const std::vector<d
 HcalTriggerPrimitiveAlgo::~HcalTriggerPrimitiveAlgo() {
 }
 
+void converttoGCTeta(int& eta)
+  {
+          if (eta == 29 || eta == 30 || eta == 31)
+          {
+                  eta = 18;
+          }
+          else if (eta == 32 || eta == 33 || eta == 34)
+          {
+                  eta = 19;
+          }
+          else if (eta == 35 || eta == 36 || eta == 37)
+          {
+                  eta = 20;
+          }
+          else if (eta == 38 || eta == 39 || eta == 40 || eta == 41)
+          {
+                  eta = 21;
+          }
+          else if (eta == -29 || eta == -30 || eta == -31)
+           {
+                   eta = 3;
+           }
+           else if (eta == -32 || eta == -33 || eta == -34)
+           {
+                   eta = 2;
+           }
+           else if (eta == -35 || eta == -36 || eta == -37)
+           {
+                   eta = 1;
+           }
+           else if (eta == -38 || eta == -39 || eta == -40 || eta == -41)
+           {
+                   eta = 0;
+           }
+ }
+
+void converttoGCTphi(int& phi)
+ {
+          if (phi == 1 || phi == 71)
+          {       
+                  phi = 0;
+          }
+          else if (phi == 3 || phi == 5)
+            {     
+                  phi = 1;
+            }
+          else if (phi == 7 || phi == 9)
+            {     
+                  phi = 2;
+            }
+          else if (phi == 11 || phi == 13)
+            {     
+                  phi = 3;
+            }
+          else if (phi == 15 || phi == 17)
+            {     
+                  phi = 4;
+            }
+          else if (phi == 19 || phi == 21)
+            {     
+                  phi = 5;
+            }
+          else if (phi == 23 || phi == 25)
+            {     
+                  phi = 6;
+            }
+          else if (phi == 27 || phi == 29)
+            {     
+                  phi = 7;
+            }
+          else if (phi == 31 || phi == 33)
+            {
+                  phi = 8;
+            }
+          else if (phi == 35 || phi == 37)
+            {
+                  phi = 9;
+            }
+          else if (phi == 39 || phi == 41)
+            {
+                  phi = 10;
+            }
+          else if (phi == 43 || phi == 45)
+            {
+                  phi = 11;
+            }
+          else if (phi == 47 || phi == 49)
+            {
+                  phi = 12;
+            }
+          else if (phi == 51 || phi == 53)
+             {
+                  phi = 13;
+             }
+          else if (phi == 55 || phi == 57)
+             {
+                  phi = 14;
+	     }
+          else if (phi == 59 || phi == 61)
+             {
+                  phi = 15;
+             }
+          else if (phi == 63 || phi == 65)
+             {
+                  phi = 16;
+             }
+          else if (phi == 67 || phi == 69)
+             {
+                  phi = 17;
+             }
+ }
 
```

Comment:
With some switch the above functions could be written a bit more compact/elegantly.